File: src/main/java/coza/trojanc/stonewriter/processor/ProcessedTemplate.java
package coza.trojanc.stonewriter.processor;

import coza.trojanc.stonewriter.processor.fields.ProcessedLineItem;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Charl-PC on 2016-10-11.
 */
public class ProcessedTemplate {

	private List<ProcessedLineItem> items = new ArrayList<>();

	public List<ProcessedLineItem> getItems() {
		return items;
	}

	public void setItems(List<ProcessedLineItem> items) {
		this.items = items;
	}
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/template/process/ProcessedTemplate.kt
package coza.trojanc.receipt.template.process

import coza.trojanc.receipt.template.process.fields.ProcessedLineItem

/**
 * A class representing a template that has been processed
 * @author Charl Thiem
 */
class ProcessedTemplate {

    private val items: MutableList<ProcessedLineItem> = ArrayList<ProcessedLineItem>()

    fun getItems(): List<ProcessedLineItem> {
        return items
    }

    fun setItems(items: List<ProcessedLineItem>) {
        this.items.clear();
        this.items.addAll(items);
    }

    fun addProcessedLineItem(processedLineItem: ProcessedLineItem){
        this.items.add(processedLineItem);
    }
}
